backend/pkg/connect/errors.go

```diff
@@ -9,7 +9,25 @@
 
 package connect
 
-import "errors"
+import (
+	"errors"
+
+	con "github.com/cloudhut/connect-client"
+)
 
 // ErrKafkaConnectNotConfigured indicates that Kafka connect was not configured in Console.
 var ErrKafkaConnectNotConfigured = errors.New("kafka connect not configured")
+
+// GetStatusCodeFromAPIError tries to parse given error as kafa connect
+// ApiError and returns the status code, if parsing is not possible it returns
+// a fallback error code
+func GetStatusCodeFromAPIError(err error, fallback int) int {
```

Comment:
Forward the actual error code of the response when possible

Reply:
Maybe we should check if the status code is within our expected range (200, 202, 400-503) or the likes, to ensure we are not getting non-sense back from Kafka connect. If it's not within the range we'll also return the fallback. The API reference says they will do so, but may introduce 3xx as well at some point. Just to ensure we are not bringing ourselves in trouble whenever they change something there (e.g. in schema registry they use custom error codes like `40401`).

Reply:
I only account for 400 and 500 status code since those are the errors that should be present